client/my-sites/plans/current-plan/my-plan-card/index.jsx

```diff
@@ -0,0 +1,65 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import PlanIcon from 'components/plans/plan-icon';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+const MyPlanCard = ( { action, isError, isPlaceholder, details, plan, tagLine, title } ) => {
+	const cardClassNames = classNames( 'my-plan-card', {
+		'has-action-only': action && ! details && ! isPlaceholder,
+	} );
+	const detailsClassNames = classNames( 'my-plan-card__details', {
+		'is-error': isError,
+		'is-placeholder': isPlaceholder && ! details,
+	} );
+	const actionClassNames = classNames( 'my-plan-card__action', {
+		'is-placeholder': isPlaceholder && ! action,
+	} );
+	const hasSecondaryArea = details || action || isPlaceholder;
+
+	return (
+		<Card className={ cardClassNames } compact>
+			<div className="my-plan-card__primary">
+				{ plan && (
+					<div className="my-plan-card__icon">
+						<PlanIcon plan={ plan } />
+					</div>
+				) }
+				<div className="my-plan-card__header">
```

Comment:
We could, though it's not necessary as it is for the `my-plan-card__secondary` element. In case of the secondary wrapper, it adds a separating line and some whitespace even if neither `details` nor `action` is provided (we handle this scenario for e.g. Pressable sites).

I don't anticipate a situation where we have no `title` *and* no `tagLine` so adding more code here to support an artificial scenario doesn't make sense in my opinion.